node/src/test/scala/com/wavesplatform/state/diffs/smart/predef/AddressTest.scala

```diff
@@ -23,37 +28,53 @@ class AddressTest extends PropSpec with PropertyChecks with Matchers with Transa
   }
 
   property("should calculate address from bytes") {
-    forAll(accountGen) { acc =>
-      val address = Address.fromPublicKey(acc.publicKey, chainId)
+    forAll(for {
+      account <- accountGen
+      version <- Gen.oneOf(DirectiveDictionary[StdLibVersion].all)
+    } yield (account, version)) { case (account, version) =>
+      val address = Address.fromPublicKey(account.publicKey, chainId)
       val script =
         s"""
            | let addressString = "$address"
            | let maybeAddress = addressFromString(addressString)
            | let address = extract(maybeAddress)
            | address.bytes
         """.stripMargin
-      runScript(script) shouldBe evaluated(ByteStr(Address.fromBytes(address.bytes, chainId).explicitGet().bytes))
+      runScript(script, ctxV = version) shouldBe evaluated(ByteStr(Address.fromBytes(address.bytes, chainId).explicitGet().bytes))
     }
   }
 
   property("should calculate address and return bytes without intermediate ref") {
-    forAll(accountGen) { acc =>
-      val address = Address.fromPublicKey(acc.publicKey, chainId)
+    forAll(for {
+      account <- accountGen
+      version <- Gen.oneOf(DirectiveDictionary[StdLibVersion].all)
+    } yield (account, version)) { case (account, version) =>
+      val address = Address.fromPublicKey(account.publicKey, chainId)
       val script =
         s"""
            | let addressString = "$address"
            | let maybeAddress = addressFromString(addressString)
            | extract(maybeAddress).bytes
         """.stripMargin
-      runScript(script) shouldBe evaluated(ByteStr(Address.fromBytes(address.bytes, chainId).explicitGet().bytes))
+      runScript(script, ctxV = version) shouldBe evaluated(ByteStr(Address.fromBytes(address.bytes, chainId).explicitGet().bytes))
     }
   }
 
   property("should fails on illegal bytes length") {
     val correctLength = Address.AddressLength
-    Address.fromBytes(Array.emptyByteArray) should produce ("Wrong addressBytes length")
-    Address.fromBytes(Array(1)) should produce ("Wrong addressBytes length")
-    Address.fromBytes(Array.fill(correctLength - 1)(1)) should produce ("Wrong addressBytes length")
-    Address.fromBytes(Array.fill(correctLength + 1)(1)) should produce ("Wrong addressBytes length")
+    Address.fromBytes(Array.emptyByteArray) should produce("Wrong addressBytes length")
+    Address.fromBytes(Array(1)) should produce("Wrong addressBytes length")
+    Address.fromBytes(Array.fill(correctLength - 1)(1)) should produce("Wrong addressBytes length")
+    Address.fromBytes(Array.fill(correctLength + 1)(1)) should produce("Wrong addressBytes length")
+  }
+
+  property("RIDE addressFromString V4 limit exceeding result") {
```

Comment:
Please add:
1) positive case where addressFromString V4 limit's not exceeded and address is correct
2) other error cases:
`runScript(s""" addressFromString("${"a" * 36}") """, ctxV = V4) shouldBe Right(unit)`
`runScript(s""" addressFromString("${"a" * 35}") """, ctxV = V4) shouldBe Right(unit)`